force-app/main/default/lwc/geElevateBatch/geElevateBatch.js

```diff
@@ -0,0 +1,51 @@
+import apexAddToElevateBatch from '@salesforce/apex/GE_GiftEntryController.addToElevateBatch';
+import apexCreateElevateBatch from '@salesforce/apex/GE_GiftEntryController.createElevateBatch';
+import apexRemoveFromElevateBatch from '@salesforce/apex/GE_GiftEntryController.removeFromElevateBatch';
+import PAYMENT_ELEVATE_ID from '@salesforce/schema/DataImport__c.Payment_Elevate_ID__c';
+import PAYMENT_ELEVATE_ELEVATE_BATCH_ID from '@salesforce/schema/DataImport__c.Payment_Elevate_Batch_Id__c';
+
+class ElevateBatch {
+
+    constructor() {
+        this.elevateBatchId = null;
+    }
+    
+    async add(tokenizedGift) {
+        return await this.performAdd(tokenizedGift, true);
+    }
+
+    async performAdd(tokenizedGift, retryOnFailure) {
```

Comment:
Is the retryOnFailure param ever used as false?  I think it should be removed and the single method should just be called "add"

Reply:
The `add()` method will attempt to add a gift to an Elevate Batch. If there is a failure on the first attempt, `performAdd()` is recursively called to try and add the gift to a _new_ batch. This second attempt is initiated on line 29 of the class where retryOnFailure is set to false. 

We do this because the existing elevate batch Id may no longer be valid, and without a way to know this proactively, we retry after the first failure, this time on a new elevate batch. Open to making any additional adjustments here if you see fit.

Reply:
Ok.  I think the issue is that the retry logic is inside the method (and the class/type).  I would expect the "add" method to return either success or failure, and the client/user of the object to then determine if they want to retry, which would involve requesting/creating a different ElevateBatch and then trying to call "add" on it.      It seems odd to have a Batch create a different Batch and call methods on it.  

A change here is not required for this WI, but we should revisit later.